pkg/kube/pipeline_cache.go

```diff
@@ -0,0 +1,96 @@
+package kube
+
+import (
+	"github.com/jenkins-x/jx/pkg/apis/jenkins.io/v1"
+	"github.com/jenkins-x/jx/pkg/client/clientset/versioned"
+	"github.com/jenkins-x/jx/pkg/log"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/tools/cache"
+	"sort"
+	"time"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PipelineNamespaceCache caches the pipelines for a single namespace
+type PipelineNamespaceCache struct {
+	pipelines map[string]*v1.PipelineActivity
```

Comment:
Maybe considering using a sync.Map https://golang.org/pkg/sync/#Map since the callbacks can be called concurrently to the `Pipelines` call.